nodejs/awsx/utils.ts

```diff
@@ -108,6 +108,26 @@ export function hasTrueBooleanMember(obj: any, memberName: string | number | sym
     return val === true;
 }
 
+/** @internal */
+export function getRegionFromOpts(opts: pulumi.CustomResourceOptions): pulumi.Output<aws.Region> {
+    if (opts.parent) {
+        return getRegion(opts.parent);
+    }
+
+    return getRegionFromProvider(opts.provider);
+}
+
+
+/** @internal */
+export function getRegion(res: pulumi.Resource): pulumi.Output<aws.Region> {
+    return getRegionFromProvider(res.getProvider("aws::"));
+}
+
+function getRegionFromProvider(provider: pulumi.ProviderResource | undefined) {
+    const region = provider ? (<any>provider).region : undefined;
+    return region || aws.config.region;
```

Comment:
Are we guaranteed that `.region` will be populated on a provider?  If not, is this correct?  If so, can't we be more precise here and return `provider ? provider.region : aws.config.region`?  Also, why do we ever not get back a `Provider`?  Is that just a deign hole in the SDK where we do not materialize a `Provider` instance for the default provider?

cc @pgavlin

Reply:
>Is that just a deign hole in the SDK where we do not materialize a Provider instance for the default provider?

Correct.  If there's no provider passed along, then trying to get the provider simply returns nothing back.  Later, in the engine, the right provider is found and used in that case.  But that doesn't help on the sdk side afaict.

> Are we guaranteed that .region will be populated on a provider?   I'm not certain we have that guarantee at all...